File: c/b2745.c
#include <stdio.h>
#include <string.h>

// 2745: 진법 변환

int main(void) {
	char N[36];
	int B, i, j;
	long long data;
	long long result = 0;

	scanf("%s %d", N, &B);

	for (i = 0; i < strlen(N); i++) {
		if (N[i] >= 'A' && N[i] <= 'Z') {
			data = N[i] - 'A' + 10;
		} else {
			data = N[i] - '0';
		}
		for (j = 0; j < strlen(N)-i-1; j++) {
			data *= B;
		}
		result += data;
	}
	printf("%lld", result);
}

File: python/b2745.py

# 2745: 진법 변환

inputs = input().split(" ")
N = inputs[0][::-1]
B = inputs[1]

result = 0

for i in range(len(N)):
	if N[i].isdigit():
		data = int(N[i])
	else:
		data = (ord(N[i]) - ord('A') + 10)
	for j in range(i):
		data *= int(B)
	result += data

print(result)